File: server/Models/SensorReading.cs
namespace Server.Models;

/// <summary>
/// A single environmental sample from the greenhouse sensor.
/// </summary>
public class SensorReading
{
    public Guid Id { get; set; }
    public long SequenceNumber { get; set; }
    public DateTime Timestamp { get; set; }
    public decimal Temperature { get; set; }
    public decimal Humidity { get; set; }
    public int Co2Ppm { get; set; }
}

File: client/src/app/models/sensor-reading.ts
/**
 * A single environmental sample from the greenhouse sensor.
 * Mirrors the server `SensorReading` (decimals are serialized as JSON numbers).
 */
export interface SensorReading {
  id: string;
  sequenceNumber: number;
  timestamp: string;
  temperature: number;
  humidity: number;
  co2Ppm: number;
}
